src/nupic/algorithms/anomaly.py

```diff
@@ -56,48 +56,53 @@ def computeRawAnomalyScore(activeColumns, prevPredictedColumns):
 class Anomaly(object):
   """Utility class for generating anomaly scores in different ways.
 
-  Supported modes:
-    MODE_PURE - the raw anomaly score as computed by computeRawAnomalyScore
-    MODE_LIKELIHOOD - uses the AnomalyLikelihood class on top of the raw
-        anomaly scores
-    MODE_WEIGHTED - multiplies the likelihood result with the raw anomaly score
-        that was used to generate the likelihood
+  :param slidingWindowSize: [optional] - how many elements are summed up;
+      enables moving average on final anomaly score; int >= 0
+
+  :param mode: (string) [optional] how to compute anomaly, one of:
+
+      - :const:`nupic.algorithms.anomaly.Anomaly.MODE_PURE`
+      - :const:`nupic.algorithms.anomaly.Anomaly.MODE_LIKELIHOOD`
+      - :const:`nupic.algorithms.anomaly.Anomaly.MODE_WEIGHTED`
+
+  :param binaryAnomalyThreshold: [optional] if set [0,1] anomaly score
+       will be discretized to 1/0 (1 if >= binaryAnomalyThreshold)
+       The transformation is applied after moving average is computed.
+
   """
 
 
   # anomaly modes supported
   MODE_PURE = "pure"
+  """
+  Default mode. The raw anomaly score as computed by
+  :func:`~.anomaly_likelihood.computeRawAnomalyScore`
+  """
   MODE_LIKELIHOOD = "likelihood"
+  """
+  Uses the :class:`~.anomaly_likelihood.AnomalyLikelihood` class, which models
+  probability of receiving this value and anomalyScore
+  """
   MODE_WEIGHTED = "weighted"
+  """
+  Multiplies the likelihood result with the raw anomaly score that was used to
+  generate the likelihood (anomaly * likelihood)
+  """
+
   _supportedModes = (MODE_PURE, MODE_LIKELIHOOD, MODE_WEIGHTED)
 
 
-  def __init__(self, 
-               slidingWindowSize=None, 
-               mode=MODE_PURE, 
+  def __init__(self,
+               slidingWindowSize=None,
+               mode=MODE_PURE,
                binaryAnomalyThreshold=None):
-    """
```

Comment:
We should have good doc strings for `__init__` functions. Usually the only info you need here is the parameter documentation. But I don't think we should move that to the class doc string just because of docs tooling.